core/src/com/etheller/warsmash/viewer5/handlers/w3x/simulation/CUnit.java

```diff
@@ -977,7 +984,20 @@ private void kill(final CSimulation simulation, final CUnit source) {
 			this.pathingInstance.remove();
 			this.pathingInstance = null;
 		}
-		popoutWorker(simulation);
+		if(!(this.getConstuctionProcessType()!=null&&this.getConstuctionProcessType()==ConstructionFlag.CONSUME_WORKER)){
+			popoutWorker(simulation);
+		}else{
+			if(workerInside!=null) {
```

Comment:
If we are already going to check again whether the worker inside is null, maybe we should have `ConstructionFlag.NO_WORKER` for the Undead instead? This would purely be a pedantic change; it's just a suggestion and it's not necessary. It's probably beneficial to null check the `workerInside` for possible crazy custom maps anyways.